src/_pytest/python.py

```diff
@@ -1181,19 +1181,25 @@ def _idval(
         if hook_id:
             return hook_id
 
+    modified_val = None  # empty strings are valid as val
```

Comment:
I think a better name for this variable would be `id`. Because I see `val` as the input value, and the `modified_val` here is actually the intended ID string for that value.

Reply:
thanks @bluetech will revisit this during this week. - Any other thoughts on a more appropriate name? I wouldnt want to shadow the id() builtin here